frontend/ui/screensaver.lua

```diff
@@ -377,6 +382,38 @@ function Screensaver:show(event, fallback_message)
     end
 end
 
+function Screensaver:expandSpecial(message, fallback)
+    -- Expand special character sequences in given message. Use fallback string if there is no document instance
+    -- %p percentage read, with percent sign
+    -- %c current page
+    -- %t total pages
+    -- %dt document title
+
+    local ret = message
+
+    local lastfile = G_reader_settings:readSetting("lastfile")
+    local doc = DocumentRegistry:openDocument(lastfile)
+    local instance = require("apps/reader/readerui"):_getRunningInstance()
+    if instance ~= nil then
+        local currentpage = instance.view.state.page
+        ret = string.gsub(ret, "%%c", currentpage)
```

Comment:
It's indeed also nice and simple this way. @poire-z any thoughts on this "custom" thingy vs the regular template function?

Reply:
Our current [template function](https://github.com/koreader/koreader-base/blob/bf8faaf2d7845c08ca2ca1c32568b4163816ea74/ffi/util.lua#L590-L599) (which uses the same kind of `string.gsub`) only handles %num => param #num, so it wouldn't work for this use case. So I'm fine with it this way.

But we could easily add a function `util.template2(str, arr)` (better name?) and provide it with `{ c=currentpage, t=totalpages ...}` so it does the gsub work there, and we have something shorter here :)

Reply:
I'm not sure if named replacement strings are really more user friendly than a simple %1, %2, etc., unless perhaps they were truly named.

Reply:
But the simple %1 %2 doesn't allow any random variables order (if we want to allow freedom if the template string comes from elsewhere)

Reply:
Huh? Random variable order is the entire reason I wrote that function in the first place! :-)

Reply:
Oh ok, never ever thought they were different :| (Lots of discussion in #1176 to come up with that :)

But anyway, a tad easier to remember: %c is current page, %p is percentage - than: %1 is current page, %4 is percentage :)